File: zkex/src/org/zkoss/zkex/zul/Fisheye.java
/* Fisheye.java

{{IS_NOTE
	Purpose:
		
	Description:
		
	History:
		Jul 8, 2008 11:59:54 AM , Created by jumperchen
}}IS_NOTE

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

{{IS_RIGHT
	This program is distributed under GPL Version 3.0 in the hope that
	it will be useful, but WITHOUT ANY WARRANTY.
}}IS_RIGHT
*/
package org.zkoss.zkex.zul;

import java.io.IOException;

import org.zkoss.lang.Objects;
import org.zkoss.zk.ui.Component;
import org.zkoss.zk.ui.UiException;
import org.zkoss.zul.impl.LabelImageElement;
import org.zkoss.zul.impl.XulElement;

/**
 * A fisheye item.
 * 
 * <p>Default {@link #getZclass}: z-fisheye.
 * 
 * @author jumperchen
 * @since 3.5.0
 */
public class Fisheye extends LabelImageElement implements org.zkoss.zkex.zul.api.Fisheye {
	public Fisheye() {
	}
	public Fisheye(String label, String image) {
		super(label, image);
	}

	//-- super --//
	public String getZclass() {
		return _zclass == null ? "z-fisheye" : _zclass;
	}
	
	/**
	 * This method is unsupported. Please use {@link Fisheyebar#setItemWidth(int)} instead.
	 */
	public void setWidth(String width) {
		throw new UnsupportedOperationException("readonly");
	}
	/**
	 * This method is unsupported. Please use {@link Fisheyebar#setItemHeight(int)} instead.
	 */
	public void setHeight(String height) {
		throw new UnsupportedOperationException("readonly");
	}
	
	public void beforeParentChanged(Component parent) {
		if (parent != null && !(parent instanceof Fisheyebar))
			throw new UiException("Unsupported parent for fisheye: "+parent);
		super.beforeParentChanged(parent);
	}
	/** Not childable. */
	protected boolean isChildable() {
		return false;
	}
}

File: zkex/src/archive/web/js/zkex/menu/Fisheye.js
/* Fisheye.js

	Purpose:
		
	Description:
		
	History:
		Thu May 15 11:17:24     2009, Created by kindalu

Copyright (C) 2009 Potix Corporation. All Rights Reserved.

This program is distributed under GPL Version 3.0 in the hope that
it will be useful, but WITHOUT ANY WARRANTY.
*/

zkex.menu.Fisheye = zk.$extends(zul.LabelImageWidget, {
	getZclass: function () {
		var zcls = this._zclass;
		return zcls != null ? zcls: "z-fisheye";
	},

	doMouseOver_: function (evt) {
		//not callback
		var cmp = this.getNode(),
			label = this.getSubnode("label");
			
		if (this._label != "") {
			label.style.display = "block";
			label.style.visibility = "hidden";
		}
		var meta = this.parent;
		if (meta) {
			if (!meta.active)
				meta.active = true;
			meta._fixLab(this);
		}
		zDom.cleanVisibility(label);
	},
	doMouseOut_: function (evt) {
		//not callback
		this.getSubnode("label").style.display = "none";
	},
	bind_: function () {//after compose
		this.$supers('bind_', arguments);
		var cmp=this.getNode(), 
			img=this.getSubnode("img"),
			label=this.getSubnode("label");
		zDom.disableSelection(cmp);
		
		// store the two attributes for better performance.
		this._mh = zDom.sumStyles(label, "tb", zDom.margins);
		this._mw = zDom.sumStyles(label, "lr", zDom.margins);
	}
});